app/controllers/questions_controller.rb

```diff
@@ -63,8 +63,12 @@ def update_response_sets(params)
   # PATCH/PUT /questions/1/publish
   def publish
     if @question.status == 'draft'
-      @question.publish(current_user)
-      render :show, statis: :published, location: @question
+      if @current_user.has_role?(:publisher)
```

Comment:
Could we use @current_user.publisher? in all of these controllers instead of @current_user.has_role?(:publisher) . 90% chance there will be more roles that get to publish things in the future, will make it easier to update.